src/Sulu/Bundle/WebsiteBundle/DependencyInjection/Compiler/RouteDefaultOptionsCompilerPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class RouteDefaultOptionsCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('routing.loader')) {
+            return;
+        }
+
+        if (!$container->hasDefinition('sulu_website.provider.content')) {
+            return;
+        }
+
+        // copy default route options which are set by the symfony FrameworkExtension based on the config:
+        // https://github.com/symfony/symfony/pull/31900
+        $container->getDefinition('sulu_website.provider.content')->replaceArgument(
+            7,
+            $container->getDefinition('routing.loader')->getArgument(1)
```

Comment:
Is it not possible to use `getArgument('$defaultOptions')` or something to be sure that we load no other argument instead?

Reply:
Unfortunately not. `The argument "$defaultOptions" doesn't exist.`

Reply:
Okay a functional test that the defaultOptions is correctly set is then for me more important, else in a symfony upgrade this is maybe forgotten to be updated.

Reply:
added a functional test case 🙂